mkdocs/docs/HPC/setting_up_python_virtual_environments.md

```diff
@@ -0,0 +1,262 @@
+deactivate                    # Deactivate the virtual environment
+```
+
+Next, we submit the job script:
+
+```bash
+$ qsub jobscript.pbs
+```
+
+Two files will be created in the directory where the job was submitted: `python_job_example.o[job_id]` and `python_job_example.e[job_id]`.
+The `.o` file contains the output of the job.
+
+
+## Troubleshooting
+
+### Illegal instruction error
+
+Activating a virtual environment created on a different cluster can cause issues. 
+This happens
+because the binaries in the virtual environments from cluster A might not work with the CPU architecture of cluster B.
+
+For example, if we create a virtual environment on the skitty cluster,
+
+```bash
+$ module swap cluster/skitty
+$ qsub -I
+$ python -m venv myenv
+```
+
+return to the login node by pressing CTRL+D and try to use the virtual environment:
+
+```bash
+$ source myenv/bin/activate
+$ python
+Illegal instruction (core dumped)
+```
+
+we are presented with the illegal instruction error. More info on this [here](troubleshooting.md#illegal-instruction-error)
+
+### Error: cannot open shared object file: No such file or directory
+
+There are two main reasons why this error could occur.
+
+1. you have not loaded the python module that was used to create the virtual environment.
+2. you added or removed modules while the virtual environment was activated.
+
+#### Entering a virtual environment while the python module used to create it is not active
+
+When you load a python module and use that to make a virtual environment, you need to make sure that the same module 
+is loaded when you enter the environment. This is because the virtual environment keeps a reference to the base python 
+used to create it.
+
+The following commands illustrate this issue:
+
+```bash
+$ module load Python/3.10.8-GCCcore-12.2.0  # Load a python module
+$ python -m venv myenv                      # Create a virtual environment with loaded python module
+$ module purge                              # Remove all loaded modules
+$ source myenv/bin/activate                 # Activate the virtual environment
+$ python                                    # Start python
+python: error while loading shared libraries: libpython3.10.so.1.0: cannot open shared object file: No such file or directory
+```
+
+Here, the virtual environment tries to use the python module that was loaded when the environment was created, which is no longer available.
+The solution is to load the same python module before activating the virtual environment:
+
+```bash
+$ module load Python/3.10.8-GCCcore-12.2.0  # Load the same python module
+$ source myenv/bin/activate                 # Activate the virtual environment
+```
+
+#### modifying modules while in a virtual environment
+
+You must not delete or add modules while in a virtual environment. 
+Adding and removing modules modifies the `$PATH` variable in the current shell. When activating a virtual environment,
```

Comment:
Be consistent in terminology, don't use "delete", "remove", "unload" to all mean the same time, only use "unload"

Same for "add" vs "load", only use "load"

Reply:
Fixed in [6fa2f76](https://github.com/hpcugent/vsc_user_docs/pull/639/commits/6fa2f76d3cc850a43bd5f8c9d8462d0b4f9c5a5e)